src/coreclr/jit/jitconfig.cpp

```diff
@@ -309,11 +320,30 @@ void JitConfigValues::MethodSet::destroy(ICorJitHost* host)
     m_names = nullptr;
 }
 
-static bool matchesName(const char* const name, int nameLen, bool wildcardAtEnd, const char* const s2)
+// strstr that is length-limited, this implementation is not intended to be used on hot paths
+static size_t strnstr(const char* pSrc, size_t srcSize, const char* needle, size_t needleSize)
 {
-    // 's2' must start with 'nameLen' characters of 'name'
-    if (strncmp(name, s2, nameLen) != 0)
+    for (size_t srcPos = 0; srcPos <= srcSize - needleSize; srcPos++)
```

Comment:
`assert(srcSize >= needleSize)`?

Reply:
if needleSize is bigger it will just return "not found" because it actually can be bigger

Reply:
But in that case won't the srcSize - needleSize way bigger? You might just fast return "not found" in such cases.

Reply:
Guess this is still unaddressed.

Reply:
@kunalspathak sorry didn't notice your comment - I am not sure I understand, let's say 
srcSize is `10` and needleSize is `20`, then `srcSize - needleSize` is `-10` so for-loop immediately exits and returns "not found" - am I wrong?

Reply:
I haven't looked in depth here, but `srcSize - needleSize` can never be negative, since `size_t` is unsigned.

Reply:
ah oops, it supposed to be ssize_t

Reply:
> I haven't looked in depth here, but `srcSize - needleSize` can never be negative, since `size_t` is unsigned.

^ That.